main.go

```diff
@@ -43,7 +43,11 @@ import (
 	"github.com/uber/tchannel-go"
 )
 
-var errHealthAndProcedure = errors.New("cannot specify procedure and use --health")
+var (
+	errHealthAndProcedure = errors.New("cannot specify procedure and use --health")
+	// map of caller names we do not want to be used.
+	disallowedCallerNames = map[string]bool{"tcurl": true}
```

Comment:
optional: prefer `map[T]struct{}` for sets rather than `map[T]bool` to avoid accidentally using the value inside the map rather than the `ok`.

(It's easy to accidentally do, `if v := disallowedCallerNames[name]; v { .. }`.)